File: app/src/main/java/org/metabrainz/mobile/data/sources/api/ServiceModule.java
package org.metabrainz.mobile.data.sources.api;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.android.components.ApplicationComponent;

@Module
@InstallIn(ApplicationComponent.class)
public class ServiceModule {

    @Singleton
    @Provides
    public LookupService getLookupService() {
        return MusicBrainzServiceGenerator.createService(LookupService.class, true);
    }

    @Singleton
    @Provides
    public CollectionService getCollectionService() {
        return MusicBrainzServiceGenerator.createService(CollectionService.class, true);
    }

    @Singleton
    @Provides
    public LoginService getLoginService() {
        return MusicBrainzServiceGenerator.createService(LoginService.class, false);
    }

    @Singleton
    @Provides
    public TaggerService getTaggerService() {
        return MusicBrainzServiceGenerator.createService(TaggerService.class, false);
    }


}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/ServiceModule.kt
package org.metabrainz.mobile.data.sources.api

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import org.metabrainz.mobile.data.sources.api.MusicBrainzServiceGenerator.createService
import javax.inject.Singleton

@Module
@InstallIn(ApplicationComponent::class)
class ServiceModule {

    @get:Provides
    @get:Singleton
    val lookupService: LookupService = createService(LookupService::class.java, true)

    @get:Provides
    @get:Singleton
    val collectionService: CollectionService = createService(CollectionService::class.java, true)

    @get:Provides
    @get:Singleton
    val loginService: LoginService = createService(LoginService::class.java, false)

}